packages/client/modules/userDashboard/components/OrgTeams/OrgTeams.tsx

```diff
@@ -1,49 +1,118 @@
 import OrgTeamsRow from './OrgTeamsRow'
-import {OrgTeams_organization$key} from '../../../../__generated__/OrgTeams_organization.graphql'
+import Menu from '../../../../components/Menu'
+import {MenuPosition} from '../../../../hooks/useCoords'
+import MenuItem from '../../../../components/MenuItem'
+import DashFilterToggle from '../../../../components/DashFilterToggle/DashFilterToggle'
+import {OrgTeamsQuery} from '../../../../__generated__/OrgTeamsQuery.graphql'
 
 const StyledPanel = styled(Panel)({
   maxWidth: ElementWidth.PANEL_WIDTH
 })
 
 type Props = {
-  organizationRef: OrgTeams_organization$key
+  queryRef: PreloadedQuery<OrgTeamsQuery>
 }
 
 const OrgTeams = (props: Props) => {
-  const {organizationRef} = props
-  const organization = useFragment(
+  const {queryRef} = props
+  const data = usePreloadedQuery<OrgTeamsQuery>(
     graphql`
-      fragment OrgTeams_organization on Organization {
-        id
-        isBillingLeader
-        teams {
-          id
-          ...OrgTeamsRow_team
+      query OrgTeamsQuery($orgId: ID!) {
+        viewer {
+          featureFlags {
+            canViewTeamsInDomain
+          }
+          organization(orgId: $orgId) {
+            id
+            activeDomain
+            isBillingLeader
+            name
+            teams {
+              id
+              ...OrgTeamsRow_team
+            }
+          }
+          domains {
+            id
+            organizations {
+              teams {
+                id
+                ...OrgTeamsRow_team
+              }
+            }
+          }
         }
       }
     `,
-    organizationRef
+    queryRef
   )
-  const {teams, isBillingLeader} = organization
+
+  const {togglePortal, originRef, menuPortal, menuProps} = useMenu(MenuPosition.UPPER_RIGHT)
+  const {viewer} = data
+  const {featureFlags, domains, organization} = viewer
+  const {activeDomain, teams, isBillingLeader, name} = organization ?? {}
+
+  const ALL_TEAMS_IN_ORG = `All Teams In ${name}`
+  const ALL_TEAMS_IN_DOMAIN = `All Teams In ${activeDomain}`
+  const [label, setLabel] = useState(ALL_TEAMS_IN_DOMAIN)
+
+  const teamsByDomain = useMemo(() => {
+    return domains.flatMap((domain) =>
+      domain.organizations.flatMap((organization) => organization.teams)
+    )
+  }, [domains])
+
+  const {canViewTeamsInDomain} = featureFlags
+
+  const handleMenuItemClick = (newLabel: string) => {
+    setLabel(newLabel)
+    togglePortal()
+  }
+
+  const selectedTeams =
+    (label === ALL_TEAMS_IN_DOMAIN && teamsByDomain.length ? teamsByDomain : teams) ?? []
   if (!isBillingLeader) return null
   return (
     <>
       <h1>{'Teams'}</h1>
+      {canViewTeamsInDomain && (
+        <>
+          <DashFilterToggle onClick={togglePortal} label={label} ref={originRef} value={label} />
+          {menuPortal(
+            <Menu
+              keepParentFocus
+              defaultActiveIdx={label === ALL_TEAMS_IN_DOMAIN ? 0 : 1}
```

Comment:
If we turn on the feature flag, I assume we'll be directing users to the teams within the domain, so I chose this as the default active index